Update crafting.lua with all legendary recipe items

VP-EN opened the pull request:
I updated the crafting.lua file with every available legendary recipe in the game. 
At least the ones that actually work. I named the iconic items, since that might be useful for someone?

I made small categories throughout the code, should more recipes/items be added later into the game. Figured it might be easier with some categories.

Note, this is only the legendary version of the recipes. There are still a lot more recipes in the game. But since the current command in the console is for legendary recipes (`BD.Cheats.Crafting.AddLegendaryRecipes()`), I figured I just wanted to add those. 

I hope I didn't mess anything up. Just used the in-house Github editor.

Commit:
Update crafting.lua with all legendary recipe items

I updated the crafting.lua file with every available legendary recipe in the game. 
At least the ones that actually work. I named the iconic items, since that might be useful for someone?

I made small categories throughout the code, should more recipes/items be added later into the game. Figured it might be easier with some categories.

Note, this is only the legendary version of the recipes. There are still a lot more recipes in the game. But since the current command in the console is for legendary recipes, I figured I just wanted to add those. 

I hope I didn't mess anything up. Just used the in-house Github editor.

osulli reviewed (approved):
Thanks

Commit:
Updated crafting.lua with new commands + new items

1) 
Added three new commands:
    Crafting.AddModRecipes()
    Crafting.AddMedRecipes()
    Crafting.AddGrenadeRecipes()

and integrated them to the
Crafting.AddAll()

2) 
I forgot the fact that in order to craft some of the legendary iconic weapons and quick-hacks, you'll need the recipe for the common and rare versions too.
Hopefully this little addition should fix it. 
I basically just added the other recipes to the individual iconic weapons and quickhacks that had other recipes available beyond the one(s) tagged legendary. I also added all the quickhacks, even the ones not available as legendary (some of them only goes to "Lvl3" which is "epic" quality ingame. Hopefully this wont be an issue.

Note:
I'm not sure how the crafting of some of the iconic weapons work. Since you would probably need 1 "rare" or "uncommon" version of the weapon, even in order to craft the epic version of it. You're usually given that, by the end of the quest where you're originally supposed to get it from (as a reward). And then you can use the recipes in order to upgrade it.

So I'm not sure if adding the recipes themselves would be enough. You might need to add the "base" (the rewarded) version of the weapon, in order to be able to use the iconic weapon recipes.. Anyways, that'll be for a later fix / version I guess.

ghost commented:
I did another commit. Hopefully it's not too much of a hassle to look through :).

Edit: woops, accidentally closed it.

osulli reviewed (changes requested):
Just need to update README.md with new protocols + update init.lua (if required)

Commit:
Updated crafting.lua, added few new commands

Added:
    Crafting.AddEpicRecipes()
    Crafting.AddRareRecipes()
    Crafting.AddUncommonRecipes()
    Crafting.AddCommonRecipes()

And added the corresponding items to the right categories/commands. 
I moved the "max LVL3" (epic) quickhacks from the legendary command to the "Crafting.AddEpicRecipes()" instead.

This commit should now include basically all recipes in the entire game.
Except the ones that doesn't work for some reason (they're --dashed out).

ghost commented:
I updated the crafting.lua once again.

Now all available/working recipes should be in there, sorted in the different quality categories.

Made a wrong pull request, now it should be included in this one lol

EDIT 1:
I don't know if the init.lua needs updating?? I'll test my version in game and see if the new commands work.

EDIT 2:
It seems all the commands work just fine in-game. All recipes are getting added correctly. At least that's what I can see.

My guess is that there's no need to update the init.lua, but I'm no expert in that regard.

Commit:
Updated README.md

Updated with the new commands
Don't think I forgot anything :)

osulli commented on mods/braindance_protocol/cheats/crafting.lua:
```diff
@@ -24,7 +24,7 @@ function Crafting.AddLegendaryMaterials(quantity)
 
 function Crafting.AddLegendaryRecipes()
-    local moduleName = "Add Legendary Crafting Recipies"
+    local moduleName = "Add Legendary Crafting Recipes"
```
TIL it isn't reci-pies but reci-pes
